substrate/frame/examples/single-block-migrations/src/runtime_upgrade_reference_docs.rs

```diff
@@ -0,0 +1,132 @@
+//! desired new blob.
+//!
+//! Prior to building the new runtime, don't forget to update the
+//! [`RuntimeVersion`](sp_version::RuntimeVersion).
+//!
+//! # Migrations
+//!
+//! It is often desirable to define logic to execute immediately after runtime upgrades.
+//!
+//! Self-contained pieces of logic that execute after a runtime upgrade are called "Migrations".
+//!
+//! Migrations are typically used to 'migrate' pallet storage into a new layout when the expected
+//! storage layout of a pallet changes, but they do anything including but not limited to:
+//!
+//! - Calling arbitrary pallet methods
+//! - Mutating arbitrary on-chain state
+//! - Cleaning up some old storage items that are no longer needed
+//!
+//! ## Single Block Migrations
+//!
+//! - Execute immediately and entirely at the beginning of the block following
+//! a runtime upgrade.
+//! - Are suitable for migrations which are guaranteed to not exceed the block weight.
+//! - Are simply implementations of [`OnRuntimeUpgrade`].
+//!
+//! To learn best practices for writing single block pallet storage migrations, see the
+//! [Single Block Migration Example Pallet](crate).
+//!
+//! ### Scheduling the Single Block Migrations to Run Next Runtime Upgrade
+//!
+//! Schedule migrations to run next runtime upgrade passing them as a generic parameter to your
+//! [`Executive`](frame_executive) pallet:
+//!
+//! ```ignore
+//! /// Tuple of migrations (structs that implement `OnRuntimeUpgrade`)
+//! type Migrations = (
+//! 	pallet_example_storage_migration::migrations::v1::versioned::MigrateV0ToV1,
+//! 	MyCustomMigration,
+//! 	// ...more migrations here
+//! );
+//! pub type Executive = frame_executive::Executive<
+//! 	Runtime,
+//! 	Block,
+//! 	frame_system::ChainContext<Runtime>,
+//! 	Runtime,
+//! 	AllPalletsWithSystem,
+//! 	Migrations, // <-- pass your migrations to Executive here
+//! >;
+//! ```
+//!
+//! ### Ensuring Single Block Migraiton Safety
+//!
+//! "My migration unit tests pass, so it should be safe to deploy right?"
+//!
+//! No! Unit tests execute the migration in a very simple test environment, and cannot account
+//! for the complexities of a real runtime or real on-chain state.
+//!
+//! Prior to deploying migrations, it is critical to perform additional checks to ensure that when
+//! run in our real runtime they will not brick the chain due to:
+//! - Panicing
+//! - Touching too many storage keys and resulting in an excessively large PoV
+//! - Taking too long to execute
+//!
+//! [`try-runtime-cli`](https://github.com/paritytech/try-runtime-cli) has a sub-command
+//! [`on-runtime-upgrade`](https://paritytech.github.io/try-runtime-cli/try_runtime_core/commands/enum.Action.html#variant.OnRuntimeUpgrade)
+//! which is designed to help with exactly this.
+//!
+//! Developers MUST run this command before deploying migrations to ensure they will not
+//! inadvertently result in a bricked chain.
```

Comment:
Can you maybe also link our CI files for the try-runtime checks? I think it would be helpful.